### PR TITLE
Add Delete Metric Profile API with name query parameter

### PR DESCRIPTION
**Pull request by shreyabiradar07**
## Description

This PR implements DELETE `/deleteMetricProfile` API supporting metric profile `name` query parameter

### Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [x] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [ ] New Test X
- [x] Functional testsuite: Tested with local monitoring test suite using MetricProfile REST API tests

**Test Configuration**
* Kubernetes clusters tested on:  minikube

## Checklist :dart:

- [x] Followed coding guidelines
- [x] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

Image: quay.io/shbirada/mvp_demo:delete-metric-profile

### CODE REVIEW

File: src/main/java/com/autotune/analyzer/utils/AnalyzerErrorConstants.java [hunk 1]
```diff
@@ -244,6 +245,8 @@ public DeleteMetricProfileAPI() {
             public static final String MISSING_METRIC_PROFILE_NAME_MSG = "Missing metric profile 'name' parameter";
             public static final String DELETE_METRIC_PROFILE_FROM_DB_FAILURE_MSG = "Failed to delete metric profile from DB: %s";
             public static final String DELETE_METRIC_PROFILE_FAILURE_MSG = "Failed to delete the specified metric profile data: %s";
+            public static final String DELETE_METRIC_PROFILE_ENTRY_NOT_FOUND_WITH_NAME = "KruizeMetricProfileEntry not found with metric profile name: ";
```
Comment (chandrams) [inline src/main/java/com/autotune/analyzer/utils/AnalyzerErrorConstants.java:248]: @shreyabiradar07 - Please include tests for these

File: design/KruizeLocalAPI.md [hunk 2]
```diff
@@ -30,6 +30,16 @@ Documentation still in progress stay tuned.
   - Example Request and Response
   - Invalid Scenarios
 
+- [Create Metric Profile API](#create-metric-profile-api)
+  - Introduction
+  - Example Request and Response
+  - Invalid Scenarios
+
+- [List Metric Profiles API](#list-metric-profiles-api)
```
Comment (chandrams) [inline design/KruizeLocalAPI.md:38]: @shreyabiradar07 - Update details on delete API

File: src/main/java/com/autotune/analyzer/utils/AnalyzerErrorConstants.java (hunk 1 above)
Comment (chandrams) [reply src/main/java/com/autotune/analyzer/utils/AnalyzerErrorConstants.java:248]: @shreyabiradar07 - Are tests included for delete API?

Comment (shreyabiradar07) [reply src/main/java/com/autotune/analyzer/utils/AnalyzerErrorConstants.java:248]: Tests are not yet included for delete API, for the 400 API response error status, there is no message being returned,  currently debugging the issue.

File: src/main/java/com/autotune/analyzer/performanceProfiles/MetricProfileCollection.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        LOGGER.info(KruizeConstants.MetricProfileConstants.ADDING_METRIC_PROFILE + "{}", metricProfileName);
+
+        if(metricProfileCollection.containsKey(metricProfileName)) {
+            LOGGER.error(KruizeConstants.MetricProfileConstants.METRIC_PROFILE_ALREADY_EXISTS + "{}", metricProfileName);
+        } else {
+            LOGGER.info(KruizeConstants.MetricProfileConstants.METRIC_PROFILE_ADDED + "{}", metricProfileName);
+            metricProfileCollection.put(metricProfileName, metricProfile);
+        }
+    }
+}
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/analyzer/performanceProfiles/MetricProfileCollection.java:75]: EOF missing

Comment (shreyabiradar07) [reply src/main/java/com/autotune/analyzer/performanceProfiles/MetricProfileCollection.java]: done

File: design/KruizeLocalAPI.md (hunk 2 above)
Comment (shreyabiradar07) [reply design/KruizeLocalAPI.md:38]: Updated KruizeLocalAPI.md with DeleteMetricProfile API details

**Review by msvinaykumar — APPROVED** [review]
Review Checklist

#### Common Review
- [ ] Code formatting adheres to the project’s style guide.
- [ ] No unnecessary commented-out code.
- [ ] Appropriate use of comments and documentation.

#### Logical Flow Review
- [ ] Code logic is clear and easy to follow.
- [ ] All branches and conditions are covered and tested.
- [ ] No redundant or duplicate code.

#### Demo Update
- [ ] Demonstration examples are updated to reflect changes.
- [ ] README and documentation include updated demo instructions.
- [ ] Any necessary demo files are included and tested.

#### Naming Conventions
- [ ] Variable and function names follow the project's naming conventions.
- [ ] File names are descriptive and adhere to naming standards.
- [ ] No ambiguous or misleading names.

#### Workflow
- [ ] CI/CD pipelines are updated and passing.
- [ ] Tests are included and all pass.
- [ ] The workflow for deploying or integrating changes is clear and documented.

#### Does It Affect ROS 
- [ ] Changes are reviewed for compatibility with ROS.
- [ ] ROS-specific configurations are updated if necessary.
- [ ] Any ROS dependencies or integrations are tested and verified.

**Comment by msvinaykumar** [comment]
LGTM

**Comment by msvinaykumar** [comment]
@shreyabiradar07 can you please confirm about http status code retun changes

**Comment by shreyabiradar07** [comment]
> @shreyabiradar07 can you please confirm about http status code retun changes

For the positive case - passing a valid metric profile name as query parameter the response status code is 200

Referring to MDN web docs - https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE#responses

> 200 (OK) status code if the action has been enacted and the response message includes a representation describing the status.

For invalid or non-existing metric profile query parameters it currently returns 400 status code

File: src/main/java/com/autotune/analyzer/utils/AnalyzerErrorConstants.java (hunk 1 above)
Comment (chandrams) [reply src/main/java/com/autotune/analyzer/utils/AnalyzerErrorConstants.java:248]: Please raise a separate PR for delete API tests.

**Comment by msvinaykumar** [comment]
> > @shreyabiradar07 can you please confirm about http status code retun changes
> 
> For the positive case - passing a valid metric profile name as query parameter the response status code is 200
> 
> Referring to MDN web docs - https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE#responses
> 
> > 200 (OK) status code if the action has been enacted and the response message includes a representation describing the status.
> 
> For invalid or non-existing metric profile query parameters it currently returns 400 status code

Ok this is cosmetic fix , Remaining LGTM so we can merge this.

**Review by msvinaykumar — APPROVED** [review]
LGTM

**Comment by msvinaykumar** [comment]
@shreyabiradar07 create a backlog for 400 error message

**Review by chandrams — APPROVED** [review]
LGTM

**Comment by chandrams** [comment]
@shreyabiradar07 - Please resolve the conflicts

**Review by chandrams — APPROVED** [review]
LGTM

**Pull request merged**